### PR TITLE
Ttt 4625

### PR DESCRIPTION
**Pull request by ComanVlad**

### CODE REVIEW

File: src/main/java/uk.co.telegraph/id/idGenerator.java [hunk 1]
```diff
@@ -0,0 +1,12 @@
+package uk.co.telegraph.id;
+
+import org.springframework.stereotype.Component;
+
+import java.util.UUID;
+
+@Component
+public class idGenerator {
```
Comment (adicontiu) [inline src/main/java/uk.co.telegraph/id/idGenerator.java]: idGenerator  -> IdGenerator

File: src/main/java/uk.co.telegraph/service/PartnerDataProviderService.java [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+import uk.co.telegraph.storage.PartnerDataProviderRepository;
+
+import java.io.IOException;
+
+@Service
+public class PartnerDataProviderService {
+
+    private final PartnerDataProviderRepository partnerDataProviderRepository;
+
+    /**
```
Comment (adicontiu) [inline src/main/java/uk.co.telegraph/service/PartnerDataProviderService.java]: You can remove these javadoc.

File: src/main/java/uk.co.telegraph/service/PartnerDataProviderService.java [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+     *
+     * @param partnerDataProviderRepository
+     */
+    public PartnerDataProviderService(PartnerDataProviderRepository partnerDataProviderRepository) {
+        this.partnerDataProviderRepository = partnerDataProviderRepository;
+    }
+
+    /**
```
Comment (adicontiu) [inline src/main/java/uk.co.telegraph/service/PartnerDataProviderService.java]: You can remove these javadoc.

File: src/main/java/uk.co.telegraph/service/PartnerDataProviderService.java [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+     *
+     * @param partnerData
+     * @return
+     */
+    public PartnerDataProvider createPartnerDataProvider(PartnerDataProvider partnerData) {
```
Comment (adicontiu) [inline src/main/java/uk.co.telegraph/service/PartnerDataProviderService.java]: you can rename createPartnerDataProvider to create as we are only handling PartnerDataProvider.

File: src/main/java/uk.co.telegraph/service/PartnerTypeService.java [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+
+    public PartnerTypeService(PartnerTypeRepository partnerTypeRepository) {
+        this.partnerTypeRepository = partnerTypeRepository;
+    }
+
+    /**
+     * Checks if the list is empty, after the call from repository
+     * @return A list with all partners
+     */
+    public List<PartnerType> getAllPartnerTypes()  {
```
Comment (adicontiu) [inline src/main/java/uk.co.telegraph/service/PartnerTypeService.java]: getAllPartnerTypes -> getAll

File: src/main/java/uk.co.telegraph/service/PartnerTypeService.java [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+            throw new PartnerTypeNotFound("Partner types were not found");
+        return partnerTypes;
+    }
+
+    /**
+     * Validates the id and calls the method from repository
+     * @param partnerTypeId
+     * @return
+     */
+    public PartnerType getPartnerType(String partnerTypeId)  {
```
Comment (adicontiu) [inline src/main/java/uk.co.telegraph/service/PartnerTypeService.java]: getPartnerType -> getById

File: src/main/java/uk.co.telegraph/storage/PartnerDataProviderRepository.java [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+    public PartnerDataProvider savePartner(PartnerDataProvider partnerDataProvider) {
+        ObjectMapper mapper= new ObjectMapper();
+        IndexRequest indexRequest = new IndexRequest(indexAlias,indexType,partnerDataProvider.getId());
+        indexRequest.opType(DocWriteRequest.OpType.CREATE);
+
+        try {
+            String jsonValue = mapper.writeValueAsString(partnerDataProvider);
+            indexRequest.source(jsonValue, XContentType.JSON);
+            IndexResponse response = esclient.index(indexRequest, RequestOptions.DEFAULT);
+            //String value=mapper.writeValueAsString(partner);
```
Comment (adicontiu) [inline src/main/java/uk.co.telegraph/storage/PartnerDataProviderRepository.java]: Check that the response is successful.

File: src/main/java/uk.co.telegraph/storage/PartnerDataProviderRepository.java [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+    private String indexAlias;
+    private String indexType;
+
+    @Autowired
+    public PartnerDataProviderRepository(ElasticSearchConfig partnerDataProviderEsConfig){
+        this.indexAlias=partnerDataProviderEsConfig.getIndexAlias();
+        this.indexType=partnerDataProviderEsConfig.getType();
+        this.esclient=new EsClientProvider(partnerDataProviderEsConfig).getClient();
+    }
+    public PartnerDataProvider savePartner(PartnerDataProvider partnerDataProvider) {
```
Comment (adicontiu) [inline src/main/java/uk.co.telegraph/storage/PartnerDataProviderRepository.java]: savePartner -> save. We might create a generic service interface which defines all the methods which can be implemented by a service, like getAll, getById, save. Please rename the methods from the PartnerTypeRepository as well.

File: src/main/java/uk.co.telegraph/storage/PartnerTypeRepository.java [hunk 9]
```diff
@@ -0,0 +1,98 @@
...
+        try {
+            GetResponse response = esClient.get(request, RequestOptions.DEFAULT);
+            ObjectMapper mapper = new ObjectMapper();
+            if (response.getSourceAsString() == null) {
+                throw new PartnerTypeNotFound(String.format("Partner type with id: %s was not found.", id));
+            }
+            return mapper.readValue(response.getSourceAsString(), PartnerType.class);
+        } catch (IOException e) {
+            log.error(e.getMessage());
+            throw new ElasticSearchPartnerTypeException("Error occurred on getting partner by id", e);
```
Comment (adicontiu) [inline src/main/java/uk.co.telegraph/storage/PartnerTypeRepository.java]: Please specify for which id the call has failed.

File: src/main/java/uk.co.telegraph/storage/converter/Converter.java [hunk 10]
```diff
@@ -0,0 +1,42 @@
...
+import org.springframework.stereotype.Component;
+import uk.co.telegraph.exception.PartnerTypeSerializationException;
+import uk.co.telegraph.storage.model.EsDocument;
+
+import uk.co.telegraph.TypeEntity;
+
+import java.io.IOException;
+
+@Component
+public abstract class Converter<T extends TypeEntity, U extends EsDocument> {
```
Comment (adicontiu) [inline src/main/java/uk.co.telegraph/storage/converter/Converter.java]: This is not used. EsDocument and TypeEntity as well. Once we make the implementation generic, we should be using these.

File: src/main/java/uk.co.telegraph/storage/model/FullPlaceDocument.java [hunk 11]
```diff
@@ -0,0 +1,20 @@
...
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import lombok.Setter;
+
+@JsonInclude(JsonInclude.Include.NON_DEFAULT)
+@Getter
+@Setter
+@NoArgsConstructor
+@AllArgsConstructor
+
```
Comment (adicontiu) [inline src/main/java/uk.co.telegraph/storage/model/FullPlaceDocument.java]: Please remove empty lines.

File: src/main/java/uk.co.telegraph/validation/PartnerTypeValidator.java [hunk 12]
```diff
@@ -0,0 +1,31 @@
...
+public class PartnerTypeValidator {
+
+
+    public static void validateId(String id){
+        validateStringDefined(id,"Id must be defined");
+    }
+
+    public void validateLabel(String label){validateStringDefined (label , "The partnerType needs to be defined");}
+
+    public void validateLabel(PartnerType partner){validateLabel(partner.getLabel());}
```
Comment (adicontiu) [inline src/main/java/uk.co.telegraph/validation/PartnerTypeValidator.java]: This is not used.

File: src/main/java/uk.co.telegraph/service/PartnerTypeService.java [hunk 13]
```diff
@@ -0,0 +1,39 @@
...
+    }
+
+    /**
+     * Checks if the list is empty, after the call from repository
+     * @return A list with all partners
+     */
+    public List<PartnerType> getAllPartnerTypes()  {
+        List<PartnerType> partnerTypes = partnerTypeRepository.getAll();
+        if (partnerTypes.isEmpty())
+            throw new PartnerTypeNotFound("Partner types were not found");
```
Comment (moraruv-tmg) [inline src/main/java/uk.co.telegraph/service/PartnerTypeService.java]: Is this an exceptional case? If we don't have any partner types, why not just return the empty list?

File: src/main/java/uk.co.telegraph/storage/PartnerTypeRepository.java [hunk 14]
```diff
@@ -0,0 +1,98 @@
...
+        SearchSourceBuilder searchSourceBuilder = new SearchSourceBuilder();
+        searchSourceBuilder.query(QueryBuilders.matchAllQuery());
+        searchRequest.indices(indexName).source(searchSourceBuilder);
+        try {
+            SearchResponse searchResponse = esClient.search(searchRequest, RequestOptions.DEFAULT);
+
+            ObjectMapper mapper = new ObjectMapper();
+            if(searchResponse==null)
+                throw new ElasticSearchPartnerTypeException("Error on getting partner types");
+            return Arrays.stream(searchResponse.getHits().getHits())
```
Comment (moraruv-tmg) [inline src/main/java/uk.co.telegraph/storage/PartnerTypeRepository.java]: Try extracting the transformation between a SearchResponse to a List<PartnerType> into a separate method. Try-catch within try-catch should be a sign that a method is trying to do too much and has a readability cost.

**Pull request merged**